test/e2e-go/features/transactions/proof_test.go

```diff
@@ -120,6 +121,19 @@ func TestTxnMerkleProof(t *testing.T) {
 	blk, err := client.BookkeepingBlock(confirmedTx.ConfirmedRound)
 	a.NoError(err)
 
+	// double check that our transaction is in the block
+	s, _, err := blk.DecodeSignedTxn(blk.Payset[proofresp.Idx])
+	a.NoError(err)
+
+	if s.ID().String() != txid.String() {
+		t.Log(txid.String())
+		for i := 0; i < len(blk.Payset); i++ {
+			s, _, _ := blk.DecodeSignedTxn(blk.Payset[i])
+			t.Log(s.Txn.ID())
+		}
+		t.FailNow()
```

Comment:
I think that the intent here is correct - but you should log more useful information that would help you to debug the issue further.
In particular, I'd suggest dumping the entire `confirmedTx` ( which includes the block number ), as well as the entire block.

Reply:
thanks for the advice